### PR TITLE
Add sample code of StreamingAnalyzeContent for CCAI Transcription

### PR DESCRIPTION
**Pull request by rogers140**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample

### CODE REVIEW

**Review by leahecole — CHANGES_REQUESTED** [review]
Hi - a couple of things - 
1) I think starting the region tag with `dialogflow` will fix the snippet-bot issue 
2) PTAL at the [Authoring guide](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#testing-guidelines) - these new samples need to include tests and type annotations

I'm marking this as a draft for now, please feel free to convert back to full PR and request another review once tests have been added! @nicain if you could pretty please also triage this so someone from the DEE side take a look at this for a product review while @rogers140 addresses my style concerns that would be awesome

**Comment by rogers140** [comment]
> Hi - a couple of things -
> 
> 1. I think starting the region tag with `dialogflow` will fix the snippet-bot issue
> 2. PTAL at the [Authoring guide](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#testing-guidelines) - these new samples need to include tests and type annotations
> 
> I'm marking this as a draft for now, please feel free to convert back to full PR and request another review once tests have been added! @nicain if you could pretty please also triage this so someone from the DEE side take a look at this for a product review while @rogers140 addresses my style concerns that would be awesome

Thanks Leah, I will address the style issues. For the tag, you mean I should add "dialogflow_" prefix in the [START.. and [END... ?

As for the test, I can add a test for the new functions added in participant_management.py, but for streaming_transcripts.py, it's communicating with microphones and I'm not sure if it's possible to add tests.

**Comment by rogers140** [comment]
So I've added some tests for the functions I introduced.

The streaming_transcription.py is a main function which I provided to customer to play with.

**Comment by rogers140** [comment]
Also the python 3.8 failure seems not related to my change.

File: dialogflow/participant_management.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 
-# Copyright 2021 Google LLC
+# Copyright 2023 Google LLC
```
[the remark this answers is not in the document]
Comment (leahecole) [reply dialogflow/participant_management.py:3]: we actually preserve copyright year when it's only an update, not a new file - I will commit this directly

**Review by leahecole — DISMISSED** [review]
Small style nits!

- In participant_management.py, it seems like a lot of f-strings got changed to .format. If it was you who made that change, please reset them to f-strings - that is now the preferred Python style. If it was somehow and automated post processor, please let us know so we can fix that, and sorry!
- 
Some commits that I made for you
- For future reference, per OSPO lawyercat rules, copyright years don't change when we're just updating files - they are always the year the file was first created 😄 
- always make sure to pin exact versions of dependencies pretty please!

**Comment by rogers140** [comment]
> Small style nits!
> 
> * In participant_management.py, it seems like a lot of f-strings got changed to .format. If it was you who made that change, please reset them to f-strings - that is now the preferred Python style. If it was somehow and automated post processor, please let us know so we can fix that, and sorry!
> 

Sure thing, just reset it and fixed some other existing places.

> Some commits that I made for you
> 
> * For future reference, per OSPO lawyercat rules, copyright years don't change when we're just updating files - they are always the year the file was first created 😄
> * always make sure to pin exact versions of dependencies pretty please!

File: dialogflow/participant_management.py [hunk 2]
```diff
@@ -175,4 +174,59 @@ def request_generator(audio_config, audio_file_path):
...
+
+# [START dialogflow_analyze_content_audio_stream]
+def analyze_content_audio_stream(
+    conversation_id,
+    participant_id,
+    sample_rate_herz,
+    stream,
+    timeout,
+    language_code,
+    single_utterance=False,
```
Comment (m-strzelczyk) [inline dialogflow/participant_management.py:188]: Please add type suggestions. Like:

```
conversation_id: int,
participant_id: int,
sample_rate_herz: int,
stream: ???
```

Is there some specific class that is a good Stream candidate?

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

File: dialogflow/participant_management.py (hunk 2 above)
Comment (rogers140) [reply dialogflow/participant_management.py:188]: Is it a new requirement? I don't see this enforcement in other places.

As for the stream, I'm afraid there is no good class candidate.

File: dialogflow/participant_management.py [hunk 3]
```diff
@@ -175,4 +174,59 @@ def request_generator(audio_config, audio_file_path):
...
+    """Stream audio streams to Dialogflow and receive transcripts and
+    suggestions.
+
+    Args:
+        conversation_id: Id of the conversation.
+        participant_id: Id of the participant.
+        sample_rate_herz: herz rate of the sample.
+        stream: the stream to process.
+        timeout: the timeout of one stream.
+        language_code: the language code of the audio
```
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply dialogflow/participant_management.py:199]: That was meant only for the `language_code` line

File: dialogflow/participant_management.py (hunk 2 above)
Comment (m-strzelczyk) [reply dialogflow/participant_management.py:188]: It's not a hard requirement, more like good practice :) We try to add this to samples we edit and prefer to have them in new samples.

Comment (m-strzelczyk) [reply dialogflow/participant_management.py:188]: For the stream type, I guess we just skip type suggestion. There is a way to do it using `typing` module, but it's quite verbose and would make the sample less readable. Just make sure to mention in docstring that `stream` has to have `generator()` method that yields bytes.

File: dialogflow/analyze_content_stream_test.py [hunk 4]
```diff
@@ -73,13 +73,42 @@ def participant_id(conversation_id):
...
+        participant_id=participant_id,
+        sample_rate_herz=16000,
+        stream=stream_generator(AUDIO_FILE_PATH),
+        language_code="en-US",
+        timeout=300
+    )
+    for response in responses:
+        print(response)
+    out, _ = capsys.readouterr()
+    assert "book a room" in out.lower()
```
[the remark this answers is not in the document]
Comment (rogers140) [reply dialogflow/analyze_content_stream_test.py:114]: done.

File: dialogflow/participant_management.py (hunk 2 above)
Comment (rogers140) [reply dialogflow/participant_management.py:190]: Sounds good. Done

File: dialogflow/participant_management.py [hunk 5]
```diff
@@ -175,4 +174,59 @@ def request_generator(audio_config, audio_file_path):
...
+    single_utterance=False,
+):
+    """Stream audio streams to Dialogflow and receive transcripts and
+    suggestions.
+
+    Args:
+        conversation_id: Id of the conversation.
+        participant_id: Id of the participant.
+        sample_rate_herz: herz rate of the sample.
+        stream: the stream to process.
```
[the remark this answers is not in the document]
Comment (rogers140) [reply dialogflow/participant_management.py]: done.

File: dialogflow/participant_management.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rogers140) [reply dialogflow/participant_management.py]: done

**Review by m-strzelczyk — APPROVED** [review]

File: dialogflow/analyze_content_stream_test.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (telpirion) [reply dialogflow/analyze_content_stream_test.py]: I need to chime in here. Printing output and using `capsys` IS a best practice.

See:
https://googlecloudplatform.github.io/samples-style-guide/#result

[the remark this answers is not in the document]
Comment (rsamborski) [reply dialogflow/analyze_content_stream_test.py]: I believe we should update the sample-style-guide as moving away from printing output was a big part of our last fixit (will share the link via internal chat).

**Review by telpirion — APPROVED** [review]

**Review by rsamborski — APPROVED** [review]

**Pull request merged**